lib/DBus.hs

```diff
@@ -252,17 +253,19 @@ receivedMessageBody (ReceivedMethodError _ msg) = methodErrorBody msg
 receivedMessageBody (ReceivedSignal _ msg) = signalBody msg
 receivedMessageBody (ReceivedUnknown _ msg) = unknownMessageBody msg
 
--- | Convert a 'Message' into a 'Char8.ByteString'. Although unusual, it is
--- possible for marshaling to fail; if this occurs, an error will be
--- returned instead.
-marshal :: Message msg => Endianness -> Serial -> msg -> Either MarshalError Char8.ByteString
+-- | Convert a 'Message' into a 'Char8.ByteString' along with all 'Fd' values
+-- mentioned in the message (the marshaled bytes will contain indices into
+-- this list). Although unusual, it is possible for marshaling to fail; if this
+-- occurs, an error will be returned instead.
+marshal :: Message msg => Endianness -> Serial -> msg -> Either MarshalError (Char8.ByteString, [Fd])
```

Comment:
Is there a way to avoid changing public API functions? Maybe add `marshalWithFds` here and in all other methods?

I prefer to avoid breaking existing API and releasing a new major version.

Reply:
Thanks for reviewing @rblaze 

I have pushed a commit to make the API backwards compatible in general (except for internal modules).

The library _will_ change behaviour in a backwards incompatible way when it comes to file descriptors, of course. If someone was calling `marshal` and `unmarshal` directly and relying on the previous behaviour then their code will now break (likely with an exception during `unmarshal` when the list of file descriptors is not supplied but is now expected). However, I think the previous behaviour was clearly not how dbus is intended to work, and people relying on a bug should probably not be surprised when their code breaks :wink: